src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/config/overrides/ConfigOverridesApplier.java

```diff
@@ -0,0 +1,117 @@
+      LOG.debug("Last updated(writer): {}, Last applied(reader): {}", lastUpdatedTimestamp,
+          lastAppliedTimestamp);
+      if (lastUpdatedTimestamp > lastAppliedTimestamp) {
+        apply(ConfigOverridesHelper.deserialize(overridesJson));
+      } else {
+        LOG.debug("Not applying override. Last updated timestamp {} is behind last applied "
+            + "timestamp {}", lastUpdatedTimestamp, lastAppliedTimestamp);
+      }
+    } catch (IOException ioe) {
+      LOG.error("Unable to deserialize overrides JSON:" + overridesJson, ioe);
+    }
+  }
+
+  private void apply(final ConfigOverrides overrides) {
+    if (PerformanceAnalyzerApp.getRcaController() != null && PerformanceAnalyzerApp.getRcaController().isRcaEnabled()) {
+      LOG.info("Applying overrides: {}", overrides.getEnable().getRcas());
+      RcaConf rcaConf = PerformanceAnalyzerApp.getRcaController().getRcaConf();
+      if (rcaConf != null) {
+        Set<String> currentMutedRcaSet = new HashSet<>(rcaConf.getMutedRcaList());
+        Set<String> currentMutedDeciderSet = new HashSet<>(rcaConf.getMutedDeciderList());
+        Set<String> currentMutedActionSet = new HashSet<>(rcaConf.getMutedActionList());
+        // check and remove any nodes that are in the disabled list that were enabled just now.
+        if (overrides.getEnable() != null) {
+          if (overrides.getEnable().getRcas() != null) {
+            currentMutedRcaSet.removeAll(overrides.getEnable().getRcas());
+          }
+          if (overrides.getEnable().getDeciders() != null) {
+            currentMutedDeciderSet.removeAll(overrides.getEnable().getDeciders());
+          }
+          if (overrides.getEnable().getActions() != null) {
+            currentMutedActionSet.removeAll(overrides.getEnable().getActions());
+          }
+        }
+
+        // union the remaining already disabled nodes with the new set of disabled nodes.
+        if (overrides.getDisable() != null) {
+          if (overrides.getDisable().getRcas() != null) {
+            currentMutedRcaSet.addAll(overrides.getDisable().getRcas());
+          }
+          if (overrides.getDisable().getDeciders() != null) {
+            currentMutedDeciderSet.addAll(overrides.getDisable().getDeciders());
+          }
+          if (overrides.getDisable().getActions() != null) {
+            currentMutedActionSet.addAll(overrides.getDisable().getActions());
+          }
+        }
+        rcaConf.updateRcaConf(currentMutedRcaSet, currentMutedDeciderSet, currentMutedActionSet);
+        setLastAppliedTimestamp(System.currentTimeMillis());
+      }
+    }
+  }
+
+  private boolean valid(final String overridesJson, final String timestamp) {
+    if (overridesJson == null || timestamp == null) {
+      return false;
+    }
+
+    if (overridesJson.isEmpty() || timestamp.isEmpty()) {
+      return false;
+    }
+
+    return NumberUtils.isCreatable(timestamp);
+  }
+
+  public long getLastAppliedTimestamp() {
```

Comment:
The callers of this are the test methods. Let's add a `VisibleForTesting` annotation.

Reply:
done.